src/c_api/c_api_symbolic.cc

```diff
@@ -1199,3 +1200,73 @@ int MXShallowCopySymbol(SymbolHandle src, SymbolHandle* out) {
+                         NDArrayHandle* in_args_handle,
+                         const mx_uint num_options,
+                         const char** keys,
+                         const char** vals) {
+  nnvm::Symbol *s = new nnvm::Symbol();
+  API_BEGIN();
+  nnvm::Symbol *sym = static_cast<nnvm::Symbol *>(sym_handle);
+  *s = sym->Copy();
+  nnvm::Graph g = Symbol2Graph(*s);
+  if (len) {
+    NDArray **in_args_ptr = reinterpret_cast<NDArray**>(in_args_handle);
+    Context default_ctx = in_args_ptr[0]->ctx();
+    mxnet::ShapeVector arg_shapes(len);
+    nnvm::DTypeVector arg_dtypes(len);
+    StorageTypeVector arg_stypes(len);
+    for (mx_uint i = 0; i < len; i++) {
+      const auto &in_arg = *(in_args_ptr[i]);
+      arg_shapes[i] = in_arg.shape();
+      arg_dtypes[i] = in_arg.dtype();
+      arg_stypes[i] = in_arg.storage_type();
+      CHECK(in_arg.ctx() == default_ctx)
+          << "args[" << i << "] is on context: " << in_arg.ctx()
+          << ", whereas args[0] is on context: " << default_ctx
+          << ". All args must be on the same context.";
+    }
+    const auto& indexed_graph = g.indexed_graph();
+    const auto num_forward_inputs = indexed_graph.input_nodes().size();
+    g.attrs["context"] = std::make_shared<nnvm::any>(
+        exec::ContextVector(indexed_graph.num_nodes(), default_ctx));
+    // infer shapes
+    g = exec::InferShape(std::move(g), std::move(arg_shapes), "__shape__");
+    if (g.GetAttr<size_t>("shape_num_unknown_nodes") != 0U) {
+      common::HandleInferShapeError(num_forward_inputs, indexed_graph,
+                                    g.GetAttr<mxnet::ShapeVector>("shape"));
+    }
+    // infer dtypes
+    g = exec::InferType(std::move(g), std::move(arg_dtypes), "__dtype__");
+    if (g.GetAttr<size_t>("dtype_num_unknown_nodes") != 0U) {
+      common::HandleInferTypeError(num_forward_inputs, indexed_graph,
+                                   g.GetAttr<nnvm::DTypeVector>("dtype"));
+    }
+    // infer stypes
+    g = exec::InferStorageType(std::move(g), std::move(arg_stypes), "__storage_type__");
+    if (g.GetAttr<size_t>("storage_type_num_unknown_nodes") != 0U) {
+      common::HandleInferStorageTypeError(num_forward_inputs, indexed_graph,
+                                          g.GetAttr<StorageTypeVector>("storage_type"));
+    }
+  }
+  std::vector<std::pair<std::string, std::string>> options_map;
+  for (mx_uint i = 0; i < num_options; ++i) {
+    options_map.emplace_back(keys[i], vals[i]);
+  }
+  const auto backend = mxnet::op::SubgraphBackendRegistry::Get()->GetSubgraphBackend(backend_name);
+  const auto& subgraph_prop_list = backend->GetSubgraphProperties();
+  for (auto property : subgraph_prop_list) {
+    property->PrePartition(g, options_map);
+    g.attrs["subgraph_property"] = std::make_shared<nnvm::any>(property);
+    g = ApplyPass(std::move(g), "BuildSubgraph");
+    g.attrs.erase("subgraph_property");
+    property->PostPartition(g);
+    s->outputs = g.outputs;
```

Comment:
can we move this outside the loop?

Reply:
we're looping over subgraph_prop_list, and for each property we call `g.attrs["subgraph_property"] = std::make_shared<nnvm::any>(property);`

Reply:
Sorry @rondogency, i retract that statement above. Lets move the output assignment outside of the loop

Reply:
Agreed. I will make this change.

Reply:
Done.